src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt

```diff
@@ -32,12 +39,87 @@ class UserResource(
+    description = "Create user",
+    security = [SecurityRequirement(name = "CREATE_USER")],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = CreateUserRequest::class)
+        )
+      ]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "User Information Returned",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = UserDetail::class))]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to create user information",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to create a user",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+
+  fun createUser(
+    @RequestBody @Valid createUserRequest: CreateUserRequest
+  ): UserDetail =
+    userService.createUser(createUserRequest)
+
+  @PreAuthorize("hasRole('ROLE_CREATE_USER')")
+  @DeleteMapping("/{username}")
+  @Operation(
+    summary = "Delete user",
+    description = "Delete user",
+    security = [SecurityRequirement(name = "CREATE_USER")],
+    responses = [
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to delete user",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to delete a user",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+
+  fun deleteUser(
+    @Schema(description = "Username", example = "testuser1", required = true)
+    @PathVariable @Size(max = 30, min = 1, message = "username must be between 1 and 30") username: String
+  ) =
+    userService.deleteUser(username)
 
   @PreAuthorize("hasRole('ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
   @GetMapping("/{username}")
   @Operation(
     summary = "Get specified user details",
     description = "Information on a specific user",
+    security = [SecurityRequirement(name = "MAINTAIN_ACCESS_ROLES_ADMIN")],
```

Comment:
does this actually do anything in the swagger?

Reply:
Nope!